.github/workflows/main.yml

```diff
@@ -42,37 +44,49 @@ jobs:
         if: github.event_name == 'pull_request'
         run: git fetch origin ${{ github.base_ref }}
 
-      - name: Set up Python 3.9
+      - name: Set up Python 3.10
         uses: actions/setup-python@v2
         with:
-          python-version: '3.9'
-          cache: pip
+          python-version: '3.10'
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.2.1
+        with:
+          virtualenvs-in-project: true
+
+      - name: Setup cache
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
 
       - name: Install dependencies
-        run: pip install -r requirements.txt
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install
 
-      - name: Build all modules
+      - name: Build and publish all modules
         if: github.event_name != 'pull_request'
-        run: beet build
+        run: poetry run beet -p beet-release.yaml build
+        env:
+          BEET_MODRINTH_TOKEN: ${{ secrets.MODRINTH_TOKEN }}
 
-      - name: Build modules from pull request
+      - name: Build all modules for pull request
         if: github.event_name == 'pull_request'
-        run: beet --set 'meta.pull_request_base=origin/${{ github.base_ref }}' build
+        run: poetry run beet -p beet-release.yaml build
```

Comment:
What was the reasoning being this change, is `meta.pull_request_base` not being used?